VideoLocker/src/main/java/org/edx/mobile/view/CourseHandoutFragment.java

```diff
@@ -70,48 +80,47 @@ public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
         if (!(NetworkUtil.isConnected(getActivity()))) {
-            showHandoutsOffline();
-        }
-
-        try {
-            final Bundle bundle = getArguments();
-            EnrolledCoursesResponse courseData = (EnrolledCoursesResponse) bundle
-                    .getSerializable(ENROLLMENT);
-            loadData(courseData);
-        } catch (Exception ex) {
-            logger.error(ex);
+            showErrorMessage(R.string.reset_no_network_message, FontAwesomeIcons.fa_wifi);
+        } else {
+            loadData();
         }
     }
 
-    private void loadData(EnrolledCoursesResponse enrollment) {
-        GetHandoutTask task = new GetHandoutTask(getActivity(), enrollment) {
+    private void loadData() {
+        GetHandoutTask task = new GetHandoutTask(getActivity(), courseData) {
 
             @Override
             public void onSuccess(HandoutModel result) {
-                try {
-                    if(result!=null&&(!TextUtils.isEmpty(result.handouts_html))){
-                        populateHandouts(result);
-                    }else{
-                        showEmptyHandoutMessage();
-                    }
-                } catch (Exception ex) {
-                    logger.error(ex);
-                    showEmptyHandoutMessage();
+                if (getActivity() == null) {
+                    return;
                 }
+
+                if (result != null && (!TextUtils.isEmpty(result.handouts_html))) {
+                    populateHandouts(result);
+                } else {
+                    CourseHandoutFragment.this.showErrorMessage(R.string.no_handouts_to_display,
+                            FontAwesomeIcons.fa_exclamation_circle);
+                }
+                isHandoutFetched = true;
             }
 
             @Override
             public void onException(Exception ex) {
+                if (getActivity() == null) {
+                    return;
+                }
+
+                isHandoutFetched = false;
                 logger.error(ex);
-                showEmptyHandoutMessage();
+                CourseHandoutFragment.this.showErrorMessage(R.string.no_handouts_to_display,
```

Comment:
The `CourseHandoutFragment.this.` is not strictly necessary here, as there doesn't seem to be any ambiguity.